2.16.1 Release: Fixes Changelog with relevant information (#871)
Azure/azure-cosmos-dotnet-v2: changelog.md

Because version 3 of the Azure Cosmos DB .NET SDK includes updated features and improved performance, we’ll retire version 2.x of this SDK on 31 August 2024.  As a result, you’ll need to update your SDK to version 3 by that date. We recommend following the [instructions](https://docs.microsoft.com/azure/cosmos-db/sql/migrate-dotnet-v3?tabs=dotnet-v3) to migrate to Azure Cosmos DB .NET SDK version 3.

## Release notes
### <a name="2.16.1"></a> 2.16.1

> * Improved availability for Direct +TCP mode by setting EnableTcpConnectionEndpointRediscovery to on by default.

Reviewer: ```suggestion
* Improved availability for Direct + TCP mode by setting EnableTcpConnectionEndpointRediscovery to `true` by default.
```

> * Improved availability for Direct +TCP mode by setting EnableTcpConnectionEndpointRediscovery to on by default.

Reviewer 2: ```suggestion
* Improved availability for Direct + TCP mode by setting EnableTcpConnectionEndpointRediscovery to on by default.
```

> * Improved retryWith(449) retry policy by having it retry sooner when multiple 449s are hit for the same request.

Reviewer: ```suggestion
* Improved RetryWith(HTTP 449) retry mechanics by having it retry sooner when multiple 449s are hit for the same request.
```

> * Fixed queries going to gateway when SDK is configured with Direct + TCP. Only impacts queries without aggregates or no MaxDegreeOfParallelism is set with the ServiceInterop.dll not being available.

Reviewer: Please update the Known Issues section, the table contains one issue for this, update the Mitigation column with the update to 2.16.1
